nibabel/nifti1.py

```diff
@@ -698,16 +698,26 @@ def get_data_shape(self):
         Allows for freesurfer hack for large vectors described in
         https://github.com/nipy/nibabel/issues/100 and
         http://code.google.com/p/fieldtrip/source/browse/trunk/external/freesurfer/save_nifti.m?spec=svn5022&r=5022#77
+
+        Allows for freesurfer hack for 7th order icosahedron surface described
+        in
+        https://github.com/nipy/nibabel/issues/309
+        https://code.google.com/p/fieldtrip/source/browse/trunk/external/freesurfer/load_nifti.m?r=8776#86
+        https://code.google.com/p/fieldtrip/source/browse/trunk/external/freesurfer/save_nifti.m?r=8776#50
         '''
         shape = super(Nifti1Header, self).get_data_shape()
         # Apply freesurfer hack for vector
-        if shape != (-1, 1, 1): # Normal case
+        if shape == (-1, 1, 1):
+            vec_len = int(self._structarr['glmin'])
+            if vec_len == 0:
+                raise HeaderDataError('-1 in dim[1] but 0 in glmin; '
+                                      'inconsistent freesurfer type header?')
+            return (vec_len, 1, 1)
+        # Apply freesurfer hack for ico7 surface
+        elif shape == (27307, 1, 6):
```

Comment:
I noticed this:

```
By default, mri_surf2surf will save the output as multiple 'slices'; has no effect for paint/w output format. For ico, the output will appear to be a 'volume' with Nv/R colums, 1 row, R slices and Nf frames, where Nv is the number of vertices on the surface. For icosahedrons, R=6. For others, R will be the prime factor of Nv closest to 6. Reshaping is for logistical purposes (eg, in the analyze format the size of a dimension cannot exceed 2^15). Use this flag to prevent this behavior. This has no effect when the output type is paint.
```

at https://surfer.nmr.mgh.harvard.edu/fswiki/mri_surf2surf - comment for `--noreshape`.